optimum/onnx/auto/configuration_onnx_auto.py

```diff
@@ -0,0 +1,211 @@
+# coding=utf-8
+# Copyright 2018 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Auto ONNX Config class."""
+import importlib
+import re
+import warnings
+from collections import OrderedDict
+from typing import List, Union
+
+from transformers.configuration_utils import PretrainedConfig
+from transformers.file_utils import CONFIG_NAME
+from transformers.models.auto.configuration_auto import (
+    CONFIG_MAPPING,
+    MODEL_NAMES_MAPPING,
+    AutoConfig,
+    _get_class_name,
+    _LazyConfigMapping,
+)
+from transformers.models.auto.dynamic import get_class_from_dynamic_module
+from transformers.onnx.config import OnnxConfig
+from transformers.utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+ONNX_CONFIG_MAPPING_NAMES = OrderedDict(
+    [
+        # Add configs here
+        ("albert", "AlbertOnnxConfig"),
+        ("bart", "BartOnnxConfig"),
+        ("bert", "BertOnnxConfig"),
+        ("camembert", "CamembertOnnxConfig"),
+        ("distilbert", "DistilBertOnnxConfig"),
+        ("gpt2", "GPT2OnnxConfig"),
+        ("ibert", "IBertOnnxConfig"),
+        ("longformer", "LongformerOnnxConfig"),
+        ("mbart", "MBartOnnxConfig"),
+        ("roberta", "RobertaOnnxConfig"),
+        ("t5", "T5OnnxConfig"),
+        ("xlm-roberta", "XLMRobertaOnnxConfig"),
+        ("xlm-roberta-xl", "XLMRobertaXLOnnxConfig"),
+    ]
+)
+
+
+def config_class_to_model_type(config):
+    """Converts a onnx config class name to the corresponding model type"""
+    for key, cls in ONNX_CONFIG_MAPPING_NAMES.items():
```

Comment:
A solution would also to build a "reverse" mapping `ONNX_CONFIG_TO_MODEL_TYPE` or something like that right after `ONNX_CONFIG_MAPPING_NAMES`'s instantation. That would allow to not loop all the time.

Reply:
Yes, but here we actually didn't use it for AutoOnnxConfig, I left it there to make it similar to what we have for Autoconfig in [transformers](https://github.com/huggingface/transformers/blob/master/src/transformers/models/auto/configuration_auto.py#L333)

Reply:
But indeed, it will be better to get it directly from FeaturesManager so that we don't need to change it all the time.